data/migrations/20190321183900_createQuantitiesTable.js

```diff
@@ -0,0 +1,42 @@
+exports.up = function(knex, Promise) {
+	return knex.schema.createTable("quantities", table => {
+		table.increments();
+		table
+			.integer("recipe_id")
+			.notNullable()
+			.unsigned()
+			.references("id")
+			.inTable("recipes")
+			.onDelete("CASCADE")
+			.onUpdate("CASCADE");
+		table
+			.integer("ingredient_id")
+			.notNullable()
+			.unsigned()
+			.references("id")
+			.inTable("ingredients")
+			.onDelete("CASCADE")
+			.onUpdate("CASCADE");
```

Comment:
These two foreign keys are nearly identical; do you think you could make a helper function for foreign ids? For example:
```javascript
const foreignId = (table, tableName, single) => {
  table.integer(`${single}_id`)
    .notNullable()
    .unsigned()
    .references('id')
    .on(tableName)
    .onDelete("CASCADE")
    .onUpdate("CASCADE");
};
```

Reply:
and the example usage in place of one of the long chains:
```javascript
foreignId(table, 'recipes', 'recipe');
```
and ta-da!